include/tables/XAttrTable.h

```diff
@@ -336,44 +347,42 @@ class XAttrTable : public DBTable<XAttrRowPart> {
   }
 
   XAttrMap get(Ndb* connection, Fmq* data_batch) {
-    AnyVec anyVec;
     Fmq batchedMutations;
     Fmq addAllXattrs;
 
-    for (Fmq::iterator it = data_batch->begin();
-         it != data_batch->end(); ++it) {
+    AnyVec anyVec;
+    XAttrPartKeyVec xattrPartKeyVec;
+    for (Fmq::iterator it = data_batch->begin(); it != data_batch->end(); ++it) {
       FsMutationRow row = *it;
       if (!row.requiresReadingXAttr() || !row.isXAttrOperation()) {
         continue;
       }
-
       if(row.mOperation == XAttrAddAll){
         addAllXattrs.push_back(row);
         continue;
       }
 
-      LOG_DEBUG("doRead batch for XAttr [ " + row.getXAttrName() + " ] to get its " << row.getNumParts() << " parts ");
-      for(Int16 index=0; index < row.getNumParts(); index++){
-        AnyMap pk;
-        pk[0] = row.mInodeId;
-        pk[1] = row.getNamespace();
-        pk[2] = row.getXAttrName();
-        pk[3] = index;
-        anyVec.push_back(pk);
-      }
+      addRetryKeys(row.mInodeId, row.getNamespace(), row.getXAttrName(), row.getNumParts(), anyVec, xattrPartKeyVec);
       batchedMutations.push_back(row);
     }
+    std::pair<AnyVec, XAttrPartKeyVec> xAttrKeys = std::make_pair(anyVec, xattrPartKeyVec);
+    XAttrMap xattrs;
+    int retry = 0;
+    while(!xAttrKeys.first.empty()) {
+      if(retry > 5) {
+        LOG_ERROR("cannot read xattrs");
```

Comment:
can you print more info

Reply:
changed the text a bit... not sure what else to write in there.
this should only happen in theory... if the xattr number of rows changes between to tries for 5 repeats in a row...